processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -132,31 +141,43 @@ public FilterTuning getFilterTuning()
   @Override
   public byte[] getCacheKey()
   {
-    boolean hasNull = false;
-    for (String value : values) {
-      if (value == null) {
-        hasNull = true;
-        break;
-      }
+    if (cacheKey == null) {
+      final boolean hasNull = values.stream().anyMatch(Objects::isNull);
```

Comment:
If `values` is a set, I think you could determine `hasNull` beforehand if you've already scanned values for empty/nulls, and avoid scanning again

Reply:
Hmm, this value doesn't seem necessary. Removed and added a unit test for it